### PR TITLE
Add `eval_on_control_instance` step

### PR DESCRIPTION
**Pull request by dokshina**
Added `eval_on_control_instance` for running code on control instance.

### CODE REVIEW

File: molecule/eval/converge.yml [hunk 1]
```diff
@@ -156,28 +229,99 @@
...
         that: >-
           eval_res == [
               "This snippet was evaled from string",
-              "Hello, my name is " + instance_info.instance_id + "!",
+              "Hello, my name is myapp.instance-2-joined!",
               "I am 43 seconds old",
               "--- {'other': 'cute-table'}\n...\n",
           ]
+
+# # EMPTY EVAL
```
Comment (hackallcode) [inline molecule/eval/converge.yml]: Debug?

**Review by hackallcode — DISMISSED** [review]

File: tasks/steps/eval.yml [hunk 2]
```diff
@@ -1,25 +1,6 @@
...
-        console_sock: '{{ instance_info.console_sock }}'
-        body: '{{ eval_body }}'
-        args: '{{ cartridge_eval_args }}'
-      register: eval_res_register
-
-    - name: 'Set "eval_res" fact'
-      set_fact:
-        eval_res: '{{ eval_res_register.fact | default(omit) }}'
+- import_tasks: 'blocks/eval.yml'
+  when: cartridge_eval_body is not none or cartridge_eval_file is not none
```
Comment (hackallcode) [inline tasks/steps/eval.yml]: What about to move `when: cartridge_eval_body is not none or cartridge_eval_file is not none` into  `blocks/eval.yml`?

File: tasks/steps/eval_on_control_instance.yml [hunk 3]
```diff
@@ -1,36 +1,14 @@
...
-      set_fact:
-        eval_body: >-
-          {{
-            cartridge_eval_body
-            if cartridge_eval_body is not none
-            else lookup("file", cartridge_eval_file)
-          }}
-      run_once: true
+    - import_tasks: 'blocks/eval.yml'
+      when: cartridge_eval_body is not none or cartridge_eval_file is not none
```
Comment (hackallcode) [inline tasks/steps/eval_on_control_instance.yml]: Rudiment

File: tasks/steps/eval_on_control_instance.yml [hunk 4]
```diff
@@ -1,36 +1,14 @@
...
-    - name: 'Set eval body'
-      set_fact:
-        eval_body: >-
-          {{
-            cartridge_eval_body
-            if cartridge_eval_body is not none
-            else lookup("file", cartridge_eval_file)
-          }}
-      run_once: true
+    - import_tasks: 'blocks/eval.yml'
```
Comment (hackallcode) [inline tasks/steps/eval_on_control_instance.yml]: Let's move `vars` to a place above `import_tasks`

File: tasks/steps/eval.yml [hunk 5]
```diff
@@ -1,25 +1,6 @@
...
+  vars:
```
Comment (hackallcode) [inline tasks/steps/eval.yml:4]: Let's move `vars` to a place above `import_tasks`

File: tasks/steps/eval_on_control_instance.yml (hunk 4 above)
Comment (dokshina) [reply tasks/steps/eval_on_control_instance.yml]: I don't like such a style.

File: tasks/steps/eval_on_control_instance.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+- name: 'BLOCK: Select control instance'
+  include_tasks: 'blocks/set_control_instance.yml'
+  run_once: true
+  when: not control_instance
```
Comment (hackallcode) [inline tasks/steps/eval_on_control_instance.yml:5]:
```suggestion
  when:
   - cartridge_eval_body is not none or cartridge_eval_file is not none
   - not control_instance 
```

File: tasks/steps/eval.yml [hunk 7]
```diff
@@ -1,25 +1,6 @@
...
+    eval_console_sock: '{{ instance_info.console_sock }}'
```
Comment (hackallcode) [inline tasks/steps/eval.yml]:
```suggestion
- import_tasks: 'blocks/eval.yml'
  vars:
    eval_console_sock: '{{ instance_info.console_sock }}'
  when: cartridge_eval_body is not none or cartridge_eval_file is not none
```

File: tasks/steps/eval_on_control_instance.yml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+- import_tasks: 'blocks/eval.yml'
+  vars:
+    eval_console_sock: '{{ control_instance.console_sock }}'
+  delegate_to: '{{ control_instance.name }}'
+  run_once: true
```
Comment (hackallcode) [inline tasks/steps/eval_on_control_instance.yml]:
```suggestion
  run_once: true
  delegate_to: '{{ control_instance.name }}'
```

**Review by hackallcode — DISMISSED** [review]

**Review by hackallcode — APPROVED** [review]

**Pull request merged**